src/commands/key.rs

```diff
@@ -63,29 +127,97 @@ impl Runnable for AddCmd {
     }
 }
 
-impl AddCmd {
+impl ListCmd {
     fn inner_run(&self, repo: CliOpenRepo) -> Result<()> {
-        // create new Repository options which just contain password information
-        let mut pass_opts = RepositoryOptions::default();
-        pass_opts.password = self.new_password.clone();
-        pass_opts.password_file = self.new_password_file.clone();
-        pass_opts.password_command = self.new_password_command.clone();
+        let used_key = repo.key_id();
+        let keys = repo
+            .stream_files()?
+            .inspect(|f| {
+                if let Err(err) = f {
+                    warn!("{err:?}");
+                }
+            })
+            .filter_map(Result::ok);
+
+        let mut table = table_with_titles(["ID", "User", "Host", "Created"]);
+        _ = table.add_rows(keys.map(|key: (_, KeyFile)| {
+            [
+                format!("{}{}", if used_key == &key.0 { "*" } else { "" }, key.0),
+                key.1.username.unwrap_or_default(),
+                key.1.hostname.unwrap_or_default(),
+                key.1
+                    .created
+                    .map_or(String::new(), |time| format!("{time}")),
+            ]
+        }));
+        println!("{table}");
+        Ok(())
+    }
+}
 
-        let pass = pass_opts
-            .evaluate_password()
-            .map_err(Into::into)
-            .transpose()
-            .unwrap_or_else(|| -> Result<_> {
-                Ok(Password::new()
-                    .with_prompt("enter password for new key")
-                    .allow_empty_password(true)
-                    .with_confirmation("confirm password", "passwords do not match")
-                    .interact()?)
-            })?;
+#[derive(clap::Parser, Debug)]
+pub(crate) struct RemoveCmd {
+    /// The key is to remove
+    id: String,
+}
 
-        let id = repo.add_key(&pass, &self.key_opts)?;
+impl Runnable for RemoveCmd {
+    fn run(&self) {
+        if let Err(err) = RUSTIC_APP
+            .config()
+            .repository
+            .run_open(|repo| self.inner_run(repo))
+        {
+            status_err!("{}", err);
+            RUSTIC_APP.shutdown(Shutdown::Crash);
+        };
+    }
+}
+
+impl RemoveCmd {
+    fn inner_run(&self, repo: CliOpenRepo) -> Result<()> {
+        repo.delete_key(&self.id)?;
+        info!("key {} successfully removed.", self.id);
+        Ok(())
+    }
```

Comment:
There is no safe-guarding in place, e.g. a second question as in 'Key {key_id} exists, do you really want to irrecoverably delete this key? (y/N)' with defaulting to `No`.

Reply:
That's good, and some safe-guarding against locking oneself out. But I think it's still important to add a normal feedback loop for deleting something. We could add a `--force` delete, so that feedback loop is being jumped over.

Reply:
with `forget` and `prune` we also don't have a feedback loop, so I would find it very irritating to have one here. `forget` and `prune` do have `--dry-run`, but as you specify a key id to remove, I don't see much value in adding a dry-run option here..

Reply:
To be fair, I think it's a bit different with `forget` and `prune`, as you also said, and they even have a `dry-run` option. I think it's just good design, to have a feedback loop when deleting something, and optionally make that circumventable. I would find it pretty weird to not have that, to be honest. The loop itself is also important to give feedback on if the key was found at all and what happens to it. I think the whole `key` subcommand(s) make a pretty good role model for a `tui`, actually. Where there would be probably also a feedback loop when deleting.

Reply:
https://docs.rs/dialoguer/0.11.0/dialoguer/struct.Confirm.html